src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0
+      || b <= 0
+      || c <= 0
+      || Math.max(a, b, c) >= (a + b + c - Math.max(a, b, c))
+    ) {
+      throw new Error('Posible any argument length is <= 0');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const s: number = (this.a + this.b + this.c) / 2;
```

Comment:
```suggestion
    const s: number = (this.a + this.b + this.c) / 2;
```

what is `s`? let's use better var names, (you can leave a,b, and c for side names)

Reply:
done